File: app/src/main/java/com/github/batulovandrey/urbandictionarycom/view/MainView.java
package com.github.batulovandrey.urbandictionarycom.view;

import com.github.batulovandrey.urbandictionarycom.adapter.DefinitionAdapter;

/**
 * @author Andrey Batulov on 27/10/2017
 */

public interface MainView {

    void showDataInRecycler();

    void showQueriesInListView();

    void filterText(String text);

    void setAdapterToRecycler(DefinitionAdapter definitionAdapter);

    void showToast(int resId);

    void showProgressbar();

    void hideProgressbar();
}

File: app/src/main/java/com/github/batulovandrey/unofficialurbandictionary/view/MainView.kt
package com.github.batulovandrey.unofficialurbandictionary.view

import com.github.batulovandrey.unofficialurbandictionary.adapter.DefinitionAdapter
import com.github.batulovandrey.unofficialurbandictionary.adapter.QueriesAdapter

/**
 * @author Andrey Batulov on 22/12/2017
 */

interface MainView {

    fun showDataInRecycler()

    fun showQueriesInListView()

    fun setAdapterToDefinitionsRecycler(definitionsAdapter: DefinitionAdapter)

    fun setAdapterToQueriesRecycler(queriesAdapter: QueriesAdapter)

    fun showToast(resId: Int)

    fun showProgressbar()

    fun hideProgressbar()

    fun initializeQueryToServer(query: String)

    fun showHint()
}